src/hw/aarch64.c

```diff
@@ -0,0 +1,28 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+#include <stdint.h>
+#include <math.h>
+
+/*
+ * __aarch64_nsec_per_tick --
+ *     Return the nsec/tick calculated using the cntfrq_el0 register.
+ */
+static uint64_t
+__aarch64_proc_freq_hz(void)
+{
+    uint64_t freq;
+    /*
+     * The Armv8-A documentation warns that on a WARM reset the register is set to "an
+     * architecturally UNKNOWN value". We assume the OS will take care of that scenario, but protect
+     * against div 0 out of an abundance of caution.
+     */
+    __asm__ volatile("\tmrs\t%0, cntfrq_el0\n" : "=r"(freq));
+    return (freq);
+}
+
+uint64_t __wti_hw_proc_freq_hz(void) __attribute__((weak, alias("__aarch64_proc_freq_hz")));
```

Comment:
I would think we would be able to accomplish this without relying on weak symbols.  Maybe put the __asm__ instruction in a macro in hardware.h, ifdefed for aarch?

Reasoning is that we haven't used weak symbols before - to my knowledge it requires linker support, so may limit our choice of linkers or complicate other tooling with sanitizers, etc.  if we can do without, we sidestep any such issues now and future.

Reply:
No the solution does *not* require weak symbols. I used the mechanism as a demonstration for the purpose of discussion.

No we haven't used weak symbols before, but they do confer the ability to inject behavior for the purpose of testing without recompilation which I think could be very useful (as discussed).

In terms of support all of the toolchains we support: gcc, llvm/clang, and msvc support weak symbols. The c runtimes (libc) all support the behavior so that it is possible to create intercept functions for library functions like `malloc` so the sanitizer and other tools actually rely on this mechanism for their implementation. I don't presume to predict the future, but it seems very unlikely that the concept and/or support for it would be removed. I can only imagine some alternate mechanism that provides for some kind of enhanced security, but that's idle speculation on my behalf.

Reply:
@marcbutler Hmm, I'm still wondering about this.  I understand the general benefit of injecting alternate implementations of certain functions without recompiling.  Do we think we'd use that in this case?  If so, then yes, let's use weak symbols.  If not - i.e. if it's theoretical at this point, and we might use it for injecting some different function, then maybe we could introduce weak symbols when we need it?

Said another way, if you have concrete ideas for its usage (here and maybe other places), let's use it.  Sorry if I seem like a stick in the mud, I just want to be sure before adopting a new (for us) language feature.

Reply:
@ddanderson To my mind `ext/test/fail_fs` is a very good candidate: allowing developers to inject behavior without requiring compilation. Anywhere around the os porting interface is also probably suitable.

That said: my purpose in piggy-backing the idea into this change as a way to evaluate it has been served and I will remove it.